packages/hashi/src/storage.js

```diff
@@ -0,0 +1,124 @@
+/**
+* This class offers an API-compatible replacement for localStorage and sessionStorage
+* to be used when apps are run in sandbox mode.
+*
+* For more information, see: https://developer.mozilla.org/en-US/docs/Web/API/Storage
+*/
+
+class Storage {
+    constructor() {
+        // This copy of the data does not persist, and is simply used to speed up
+        // and allow for synchronous access to data. We need to ensure that this
+        // data is populated from Kolibri before handing this off to apps if possible.
+        this.sessionData = {};
+
+        // keys have list ordering, so we use a list to ensure order is not changed.
+        // Per the spec, key ordering is defined by user agent, so we don't have to
+        // model any specific sorting behavior, just be consistent.
+        this.sessionKeys = [];
+        this.initMessageSent = false;
+        this.dataReceived = false;
+    }
+
+    _sendMessage(name, data) {
+        var message = {
+            'action': name,
+            'params': data
+        };
+        window.parent.postMessage(JSON.stringify(message), '*');
+    }
+
+    initData() {
+      var timeout = 5000;
+      var pinkyPromise = null;
+      if (!this.initMessageSent) {
+        var self = this;
+        pinkyPromise = new Promise(function(resolve, reject) {
```

Comment:
If you use arrow functions `(resolve, reject) => {}` it will bind `this` to `this` in the scope in which it is defined, and you can dispense with the `self` declaration.